src/t_pstring.c

```diff
@@ -803,12 +939,23 @@ int pmSetrangeCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
   if (offset < 0) {
     return RedisModule_ReplyWithError(ctx, "ERR offset is out of range");
   }
+  KVDKValueType type;
+  KVDKStatus s = KVDKTypeOf(engine, key_str, key_len, &type);
+  if (s == Ok && type != String)
+    return RedisModule_ReplyWithError(ctx, "Err WrongType");
   const char *val_str = RedisModule_StringPtrLen(argv[3], &val_len);
   SetRangeArgs args;
   args.val_str = val_str;
   args.val_len = val_len;
   args.ll_offset = offset;
-  KVDKStatus s = setRange(key_str, key_len, &args);
+  KVDKWriteOptions *write_option = KVDKCreateWriteOptions();
+  int64_t expire_time;
+  s = KVDKGetTTL(engine, key_str, key_len, &expire_time);
```

Comment:
pls refer to line 132

Reply:
Thanks for the review comment. I have modified the code accordingly.